packages/superset-ui-chart/src/query/Metric.ts

```diff
@@ -26,34 +26,37 @@ export enum Aggregate {
 }
 
 export enum ExpressionType {
+  BUILTIN = 'BUILTIN',
   SIMPLE = 'SIMPLE',
   SQL = 'SQL',
 }
 
-interface AdhocMetricSimple {
+interface MetricSimple {
   expressionType: ExpressionType.SIMPLE;
   column: Column;
   aggregate: Aggregate;
+  label?: string;
```

Comment:
Totally cool with this more descriptive way of listing out all the optional fields of the metrics. Alternatively, we can have something like
```
interface AdditionalMetricFields {
  label?: string;
  optionName?: string;
}
``` 
And then
```
type SimpleMetric = {
  expressionType: ExpressionType.SIMPLE;
  column: Column;
  aggregate: Aggregate;
} & Partial<AdditionalMetricFields>
```

Reply:
Actually if we use `Partial<>`, do we need to do 
```
interface AdditionalMetricFields {
  label?: string;
  optionName?: string;
}
```
instead of 
```
interface AdditionalMetricFields {
  label: string;
  optionName: string;
}
```

Reply:
Whoops, typo on my part, I meant
```
interface AdditionalMetricFields {
  label: string;
  optionName: string;
}
```
(Correcting the original comment as well)
Partial makes a subset of the keys (including all keys) optional.